Sources/StreamChat/Utils/TranslationLanguage.swift

```diff
@@ -0,0 +1,133 @@
+    public static let italian: TranslationLanguage = TranslationLanguage(languageCode: "it")
+    public static let japanese: TranslationLanguage = TranslationLanguage(languageCode: "ja")
+    public static let korean: TranslationLanguage = TranslationLanguage(languageCode: "ko")
+    public static let latvian: TranslationLanguage = TranslationLanguage(languageCode: "lv")
+    public static let malay: TranslationLanguage = TranslationLanguage(languageCode: "ms")
+    public static let norwegian: TranslationLanguage = TranslationLanguage(languageCode: "no")
+    public static let persian: TranslationLanguage = TranslationLanguage(languageCode: "fa")
+    public static let pashto: TranslationLanguage = TranslationLanguage(languageCode: "ps")
+    public static let polish: TranslationLanguage = TranslationLanguage(languageCode: "pl")
+    public static let portuguese: TranslationLanguage = TranslationLanguage(languageCode: "pt")
+    public static let romanian: TranslationLanguage = TranslationLanguage(languageCode: "ro")
+    public static let russian: TranslationLanguage = TranslationLanguage(languageCode: "ru")
+    public static let serbian: TranslationLanguage = TranslationLanguage(languageCode: "sr")
+    public static let slovak: TranslationLanguage = TranslationLanguage(languageCode: "sk")
+    public static let slovenian: TranslationLanguage = TranslationLanguage(languageCode: "sl")
+    public static let somali: TranslationLanguage = TranslationLanguage(languageCode: "so")
+    public static let spanish: TranslationLanguage = TranslationLanguage(languageCode: "es")
+    public static let spanishMexico: TranslationLanguage = TranslationLanguage(languageCode: "es-MX")
+    public static let swahili: TranslationLanguage = TranslationLanguage(languageCode: "sw")
+    public static let swedish: TranslationLanguage = TranslationLanguage(languageCode: "sv")
+    public static let tagalog: TranslationLanguage = TranslationLanguage(languageCode: "tl")
+    public static let tamil: TranslationLanguage = TranslationLanguage(languageCode: "ta")
+    public static let thai: TranslationLanguage = TranslationLanguage(languageCode: "th")
+    public static let turkish: TranslationLanguage = TranslationLanguage(languageCode: "tr")
+    public static let ukrainian: TranslationLanguage = TranslationLanguage(languageCode: "uk")
+    public static let urdu: TranslationLanguage = TranslationLanguage(languageCode: "ur")
+    public static let vietnamese: TranslationLanguage = TranslationLanguage(languageCode: "vi")
+    
+    let languageCode: String
+    
+    init(languageCode: String) {
+        self.languageCode = languageCode
+    }
+    
+    init?(locale: Locale) {
+        guard let languageCode = locale.languageCode else { return nil }
+        let regionCode = locale.regionCode ?? ""
+        self.languageCode = languageCode + (regionCode.isEmpty ? "" : "-\(regionCode)")
+    }
+    
+    static var allCases: [Self] = [
```

Comment:
This can lead to future async with the real all cases (if we forget to add a new language in this array). 

A possible solution to avoid this is to create an internal enum only for the Translation language Codes we support, and here we can the Enum -> Translation Language Code

Reply:
That's a good point, let me investigate this a bit

Reply:
> here we can the Enum -> Translation Language Code
This is not possible. we should use a code generation mechanism (SwiftGen) for this. Let's keep it out of scope for now